### ISSUE TITLE
router-vite-plugin 1.26.3 broken on Windows: "?" is not valid in a filename

### ISSUE THREAD

**Issue by msssk** [TanStack/router#1412]
### Describe the bug

The addition of:
https://github.com/TanStack/router/blob/main/packages/router-vite-plugin/src/tests/snapshots/function-declaration%3Fsplit.tsx

makes 1.26.3 not work on Windows

### Your Example Website or App

http://not.applicable.com

### Steps to Reproduce the Bug or Issue

`yarn add @tanstack/router-vite-plugin`

### Expected behavior

Package should install, but instead fails with
```
Error: While persisting /C:/Users/user/AppData/Local/Yarn/Berry/cache/@tanstack-router-vite-plugin-npm-1.25.0-d9139b073d-10c0.zip/node_modules/@tanstack/router-vite-plugin/ -> /C:/src/node_modules/@tanstack/router-vite-plugin ENOENT: no such file or directory, open 'C:\src\node_modules\@tanstack\router-vite-plugin\src\tests\snapshots\function-declaration?split.tsx'
```

### Platform

- OS: Windows
- Browser: N/A
- Version: router-vite-plugin 1.26.3

**Comment by SeanCassiere** [comment]
@tannerlinsley something to keep in mind for the `experimental.enableCodeSplitting` flag.

**Comment by msssk** [comment]
This problem still occurs with 1.26.8
`yarn add @tanstack/router-vite-plugin@1.26.8`

**Comment by SeanCassiere** [comment]
For whatever reason, till version 1.26.7, the package for `@tanstack/router-vite-plugin` ships the tests in the package bundle as seen [here](https://www.npmjs.com/package/@tanstack/router-vite-plugin?activeTab=code).

<img width="1612" alt="image" src="https://github.com/TanStack/router/assets/33615041/06002002-d398-4c56-811d-291f9a430808">

Needs further investigation as to why the build process isn't removing these files from the deployed builds.

**Comment by SeanCassiere** [comment]
The fix for this has been implemented and merged into the codebase in #1431, but it didn't trigger a release.

**Comment by SeanCassiere** [comment]
Triggered a forced release, and now the fix is available in version [1.26.10](https://github.com/TanStack/router/releases/tag/v1.26.10).

As you can see in the npm package code explorer the tests are no longer being shipped to the end user.
<img width="1364" alt="image" src="https://github.com/TanStack/router/assets/33615041/70f21fac-63a2-4498-9d62-5e5d21a41c42">
